ionic_user_interface/src/views/ViewRoute.vue

```diff
@@ -43,50 +59,59 @@
         </div>
         <br />
         <MessageBox ref="msgBox" color="danger" class="rounded margin" />
-        <ion-row
-          v-if="!hasAlreadyCommented"
-          class="ion-align-items-start ion-justify-content-start"
-        >
-          <ion-textarea
-            placeholder="Write a comment..."
-            class="ion-no-margin"
-            maxlength="150"
-            v-model="commentText"
-            autoGrow
-          ></ion-textarea>
-          <ion-button @click="postCommentHandler" fill="clear" color="dark">
-            <ion-icon :icon="sendSharp"></ion-icon>
-          </ion-button>
-        </ion-row>
-        <br />
-        <ion-card
-          v-for="({ username, timestamp, comment }, index) in routeDetails.comments"
-          :key="index"
-        >
-          <ion-card-header>
-            <ion-card-title>{{ comment }}</ion-card-title>
-            <div v-if="isLoggedIn" class="center-right">
-              <ion-icon
-                class="icon-button"
-                v-if="username !== myUsername"
-                @click="() => reportCommentHandler(username)"
-                :icon="flagOutline"
-              ></ion-icon>
-              <ion-icon
-                class="icon-button"
-                v-if="username === myUsername || isAdmin"
-                @click="() => deleteCommentHandler(username, timestamp)"
-                :icon="trashOutline"
-              ></ion-icon>
-            </div>
-          </ion-card-header>
-          <ion-card-content>
-            <ion-item class="ion-no-padding">
-              <ion-icon class="margin-right" slot="start" :icon="personCircleOutline"></ion-icon>
-              <ion-label>{{ username }}</ion-label>
-            </ion-item>
-          </ion-card-content>
-        </ion-card>
+        <div class="margin-left margin-right">
+          <ion-row
+            v-if="!hasAlreadyCommented"
+            class="ion-align-items-start ion-justify-content-start margin"
+          >
+            <ion-textarea
+              placeholder="Write a comment..."
+              class="ion-no-margin"
+              maxlength="150"
+              v-model="commentText"
+              autoGrow
+            ></ion-textarea>
+            <ion-button @click="postCommentHandler" fill="clear" color="dark">
```

Comment:
If its public who tries to post comment, can route them to log in page. Right now it does nothing

Reply:
Maybe can do the same for vote button, but its okay to leave it as it is too in case too convoluted

Reply:
I just made the comment box disappear if not logged in. Also for the vote button I made it non-clickable if not logged in